atqa-framework/tests/pages/BasePage.ts

```diff
@@ -0,0 +1,117 @@
+import { ClockUnit } from '../../utils/clockUnit';
+import { Locator } from '../../interfaces/Locator.interface';
+
+const clockUnit = new ClockUnit();
+const { I } = inject();
+
+export class BasePage {
+  public BROWSER_URLS = {
+    CHROMIUM: (id: string) => ({
+      POPUP_URI: `chrome-extension://${id}/popup.html`,
+    }),
+    GECKO: (id: string) => ({
+      POPUP_URI: `moz-extension://${id}/popup.html`,
+    }),
+    CHROME_SYSTEM: 'chrome://system/',
+    CHROMIUM_EXTENSIONS_PAGE: 'chrome://extensions/',
+    FIREFOX_EXTENSIONS_PAGE: 'about:debugging#/runtime/this-firefox',
+  };
+
+  public BROWSER_SELECTORS = {
+    CHROMIUM_EXTENSION_VALUE_BUTTON: { id: 'extensions-value-btn' },
+    CHROMIUM_EXTENSIONS_LIST: { xpath: '(//div[@class="stat-value"])[5]' },
+    CHROMIUM: {
+      DETAILS_BUTTON: {
+        shadow: [
+          'extensions-manager',
+          'extensions-item-list',
+          'extensions-item',
+          '#detailsButton',
+        ],
+      },
+      DEV_MODE_TOGGLE: {
+        shadow: ['extensions-manager', 'extensions-toolbar', '#devMode'],
+      },
+    },
+    EDGE: {
+      DETAILS_BUTTON: { xpath: '//span[contains(text(),"Details")]' },
+      DEV_MODE_TOGGLE: { css: '#developer-mode' },
+      EXTENSION_MENU_BUTTON: {
+        xpath: '(//button[@aria-label= "Extensions menu"])[2]',
+      },
+    },
+    FIREFOX: {
+      MANIFEST_EMAIL_TEXT: 'support@wavesplatform.com',
+    },
+  };
```

Comment:
Nested selectors are harder to understand.
Also, mixing them with constants is a bad idea.

Reply:
The structure isn't finished yet, maybe it is a reason for some confusion. I'm still working on it and will be more clear in the feature. But just in case, nested selectors is one the idea of POM it helps describe a page with correct and flexible structure instead declare it as a list. Selectors list do nothing, the flat structure is nicy when u are testing 1-page web application (but u don't need any pom in this way). But if your app has a complicated logic, lot of the pages and elements on it so this should be describe as structure for efficient work with that.

Reply:
As far as I know the idea of POM is to define a page object with locators and methods for interacting with it.
POM doesn't imply nesting of objects. Many pages - many POM objects.

You can see some examples here
https://www.selenium.dev/documentation/test_practices/encouraged/page_object_models/

Reply:
check my comments above